### PR TITLE
Upgrade Jackson Databind version to 2.13.4.2

### PR DESCRIPTION
**Pull request by margarit-h**
### Description
[Jackson version used by SQL plugin must match Jackson version used by [OpenSearch 2.x.](https://github.com/opensearch-project/opensearch/tree/2.x)]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ buildscript {
         opensearch_version = System.getProperty("opensearch.version", "3.0.0-SNAPSHOT")
         spring_version = "5.3.22"
         jackson_version = "2.13.4"
```
Comment (MitchellGale-BitQuill) [inline build.gradle:11]: Do we still need jackson_version?

**Review by MitchellGale-BitQuill — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (margarit-h) [reply build.gradle:11]: Yes, it is used in other Jackson dependencies such as `jackson-core` , `jackson-annotations`, `jackson-dataformat-cbor`

**Review by Yury-Fridlyand — CHANGES_REQUESTED** [review]
Few references are missed:
image

**Comment by margarit-h** [comment]
This PR is replaces by the following one which is based on `upstream/2.x`: https://github.com/Bit-Quill/opensearch-project-sql/pull/148

**Pull request closed without merging**